data/data/aws/bootstrap/main.tf

```diff
@@ -61,10 +61,12 @@ resource "aws_s3_bucket_object" "ignition" {
   server_side_encryption = "AES256"
 
   tags = merge(
+    {
+      "kubernetes.io/cluster/${var.cluster_id}" = "owned"
+    },
     {
       "Name" = "${var.cluster_id}-bootstrap"
     },
-    local.tags,
```

Comment:
We discussed this in a meeting just now. s3 objects  have a limit of 10 tags. We have two options here:

1. Skip user tags.
2. Only take the first X number of tags which would end up totaling 10. So if we have 1 static tag, we would take 9 user-defined tags.

I lean toward option 2 and I believe we are in agreement, because otherwise removing tagging would generally be a regression in the product.

Reply:
> otherwise removing tagging would generally be a regression in the product.

I talked to @TrilokGeer about this a bit and he explained the challenge of taking the "first" N tags given it's an unsorted map.  I suggested sorting the map alphanumerically and then taking the first N from the sorted list.  So that's one option to at least provide deterministic behavior when you have more than N tags provided.

But it also sounds like we shouldn't have tagged these objects in the first place, and they are ephemeral (being deleted post bootstrap).  Between those two things, I do have to wonder if it's really critical we preserve this behavior or if we can just remove the behavior and not have to add additional complexity around selecting which tags to use on the bucket objects.

I don't have much sense of why a customer would particularly care about the tagging of these resources, do you?

Reply:
There is also discussion in #6303, so the context is spread between the two PRs. Terraform already sorts map keys lexicographically, so that aspect should be handled without further effort.

I wouldn't consider maintaining this functionality to be critical, but I also don't have context for customer use cases. In general, I have gathered that customers come up with policies and automation that use tags. I don't have enough knowledge to say that those cases would never apply to s3 objects.

Hopefully this added clarification about terraform sorting clarifies the level of complexity. Given this info, which I learned yesterday, I don't think it is burdensome to implement, but I am definitely open to skipping.